lib/AST/DiagnosticEngine.cpp

```diff
@@ -1040,24 +1040,40 @@ DiagnosticBehavior DiagnosticState::determineBehavior(const Diagnostic &diag) {
 
 void DiagnosticEngine::flushActiveDiagnostic() {
   assert(ActiveDiagnostic && "No active diagnostic to flush");
+  handleDiagnostic(std::move(*ActiveDiagnostic));
+  ActiveDiagnostic.reset();
+}
+
+void DiagnosticEngine::handleDiagnostic(Diagnostic diag) {
```

Comment:
I usually try to avoid `&&` parameters unless the only right way of using the API is to move the argument, though yeah maybe in this case it's warranted, I'll update it. Thanks!